hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSink.java

```diff
@@ -96,6 +96,8 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
 
       // Append mode
       if (OptionsResolver.isAppendMode(conf)) {
+        // close compaction for append mode
+        conf.set(FlinkOptions.COMPACTION_SCHEDULE_ENABLED, false);
```

Comment:
Did you check the compaction scheduling condition:

```java
public static boolean needsScheduleCompaction(Configuration conf) {
    return OptionsResolver.isMorTable(conf)
        && conf.getBoolean(FlinkOptions.COMPACTION_SCHEDULE_ENABLED);
  }
```

Reply:
Hi @danny0405 , if the table type is mor and the compaction.schedule.enabled is true and  the write.operation is append mode,  the needsScheduleCompaction returns true.

Reply:
> Did you check the compaction scheduling condition:
> 
> ```java
> public static boolean needsScheduleCompaction(Configuration conf) {
>     return OptionsResolver.isMorTable(conf)
>         && conf.getBoolean(FlinkOptions.COMPACTION_SCHEDULE_ENABLED);
>   }
> ```

Do you mean to add checks here?

Reply:
For append mode, we mean COW + INSERT, that means `needsScheduleCompaction` returns false, and there will never schdule any compaction plans.

Reply:
OptionsResolver.isMorTable(conf) = true 
conf.getBoolean(FlinkOptions.COMPACTION_SCHEDULE_ENABLED) = ture 

Why is false?

Reply:
@danny0405